src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -410,25 +421,72 @@ internal sealed class RawArrayData
     [StructLayout(LayoutKind.Explicit)]
     internal unsafe struct MethodTable
     {
+        /// <summary>
+        /// The low WORD of the first field is the component size for array and string types.
+        /// </summary>
         [FieldOffset(0)]
         public ushort ComponentSize;
+
+        /// <summary>
+        /// The flags for the current method table (only for not array or string types).
+        /// </summary>
         [FieldOffset(0)]
         private uint Flags;
+
+        /// <summary>
+        /// The base size of the type (used when allocating an instance on the heap).
+        /// </summary>
         [FieldOffset(4)]
         public uint BaseSize;
-        [FieldOffset(0x0e)]
+
+        // See additional native members in methodtable.h, not needed here yet.
+        // 0x8: m_wFlags2 (additional flags)
+        // 0xA: m_wToken (class token if it fits in 16 bits)
+        // 0xC: m_wNumVirtuals
+
+        /// <summary>
+        /// The number of interfaces implemented by the current type.
+        /// </summary>
+        [FieldOffset(0x0E)]
         public ushort InterfaceCount;
+
+        // For DEBUG builds, there is a conditional field here (see methodtable.h again).
+        // 0x10: debug_m_szClassName (display name of the class, for the debugger)
+
+        /// <summary>
+        /// A pointer to the parent method table for the current one.
+        /// </summary>
         [FieldOffset(ParentMethodTableOffset)]
         public MethodTable* ParentMethodTable;
+
+        // Additional conditional fields (see methodtable.h).
+        // m_pLoaderModule
+        // m_pWriteableData
+        // union {
+        //   m_pEEClass (pointer to the EE class)
+        //   m_pCanonMT (pointer to the canonical method table)
+        // }
+
+        /// <summary>
+        /// This element type handle is in a union with additional info or a pointer to the interface map.
+        /// Which one is used is based on the specific method table being in used (so this field is not
+        /// always guaranteed to actually be a pointer to a type handle for the element type of this type).
+        /// </summary>
         [FieldOffset(ElementTypeOffset)]
-        public void* ElementType;
+        public void* ElementTypeHandle;
```

Comment:
This is named `ElementType` in the unmanaged MethodTable definition. I would avoid diverging the names between managed and unmanaged definitions.

Reply:
Fixed in f2ede92537c1d3e0b44b6436f686129757d97980 👍